src/main/java/graphql/language/NodeVisitor.java

```diff
@@ -0,0 +1,40 @@
+package graphql.language;
+
+public interface NodeVisitor<T> {
```

Comment:
Is there a reason why every method returns Object and not T?

Reply:
yes, there is. 
Visitor can control the Traverser if it returns some special values:
- ```TraverserMarkers.QUIT``` causes the Traverser to stop the main loop and return. Useful in the scenarios where you need to fihnd something in the tree/graph and stop searching.
- ```TraverserMarkers.ABORT``` causes the Traverser not to visit subtree if vi

Reply:
thanks for the explanation ... makes sense.

What do u think about making the methods more readable by changing it to visitArgument, vistArrayValue etc?

Reply:
@andimarek  this would severely cripple visitor pattern implementation, isn't it? Every implementation of `Node.accept()` would be required to perform reflection to figure out actual runtime type of input in order to properly dispatch next call. It cancels benefits from double-dispatch and unnecessarily complicates everything for a client code.

Reply:
@exbe but every accept method knows what type it is, why is reflection needed? just call the appropriate method. Or do I miss something?

Reply:
@andimarek you are right, yes. Accept knows which method to call on visitor. I've got biased towards method overloading :) 
I guess were are examples, where it make sense to express some-kind of grouping. 
Personally I would keep it simple and use 'visit' for root visitor.

Reply:
@exbe I would like to see it renamed and make it more explicit to improve the readability: I am not really happy for example with this:  https://github.com/graphql-java/graphql-java/pull/905/files?diff=split#diff-516be31e8c1f5179246f61a317781d1c 
I think making it explicit by naming it visitXXX makes it more readable.